tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,208 @@
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.security.JaasUtils;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * {@link StreamsCleanupClient} resets a Kafka Streams application to the very beginning. Thus, an application can
+ * reprocess the whole input from scratch.
+ * <p>
+ * This includes, setting source topic offsets to zero, skipping over all intermediate user topics,
+ * and deleting all internally created topics.
+ */
+public class StreamsCleanupClient {
+    private static OptionSpec<String> bootstrapServerOption;
+    private static OptionSpec<String> zookeeperOption;
+    private static OptionSpec<String> applicationIdOption;
+    private static OptionSpec<String> sourceTopicsOption;
+    private static OptionSpec<String> intermediateTopicsOption;
+
+    private OptionSet options = null;
+
+    public void run(String[] args) {
+        ZkUtils zkUtils = null;
+
+        int exitCode = 0;
+        try {
+            parseArguments(args);
+
+            zkUtils = ZkUtils.apply(options.valueOf(zookeeperOption),
+                    30000,
+                    30000,
+                    JaasUtils.isZkSecurityEnabled());
+
+            final List<String> allTopics = scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics());
+
+            resetSourceTopicOffsets();
+            seekToEndIntermediateTopics(zkUtils, allTopics);
+            deleteInternalTopics(zkUtils, allTopics);
+        } catch (Exception e) {
+            exitCode = -1;
+            System.err.println(e.getMessage());
+        } finally {
+            if (zkUtils != null) {
+                zkUtils.close();
+            }
+            System.exit(exitCode);
+        }
+    }
+
+    private void parseArguments(final String[] args) throws IOException {
+        final OptionParser optionParser = new OptionParser();
+        applicationIdOption = optionParser.accepts("application-id", "The Kafka Streams application ID (application.id)")
+                .withRequiredArg()
+                .describedAs("id")
+                .ofType(String.class)
+                .required();
+        bootstrapServerOption = optionParser.accepts("bootstrap-server", "Format: <host:port>")
+                .withRequiredArg()
+                .describedAs("url")
+                .defaultsTo("localhost:9092")
+                .ofType(String.class);
+        zookeeperOption = optionParser.accepts("zookeeper", "Format: <host:port>")
+                .withRequiredArg()
+                .describedAs("url")
+                .defaultsTo("localhost:2181")
+                .ofType(String.class);
+        sourceTopicsOption = optionParser.accepts("source-topics", "Comma separated list of user source topics")
```

Comment:
Assume you just wan to clear the current state, but resume where you left of (ie, no reprocessing). This would be helpful if you changed code and just want to go on where you left of.